riscv/src/lib.rs

```diff
@@ -33,7 +33,8 @@ impl RuntimeLibs {
         Self {
             arith: false,
             keccak: false,
-            poseidon: false,
+            // Poseidon is required for publics.
+            poseidon: true,
```

Comment:
I don't have full context here but if it's required, does it still make sense for it to be in this struct? Is the value ever going to be false?

Reply:
hm yea you could still want non-continuations cases without publics. I'll revert this and change the tests/APIs accordingly

Reply:
yea actually it's not only about continuations, it's mainly about publics, so it is indeed always required by the RISCV machine

Reply:
Ok removed `poseidon` from RuntimeLibs field, it's now set inside Runtime itself as default